src/TableGateway/AbstractTableGateway.php

```diff
@@ -189,32 +191,35 @@ public function select($where = null)
 
         if ($where instanceof \Closure) {
             $where($select);
-        } elseif ($where !== null) {
+        } elseif ($where !== NULL) {
             $select->where($where);
         }
 
-        return $this->selectWith($select);
+        return $this->selectWith($select, $params);
     }
 
     /**
      * @param Select $select
-     * @return null|ResultSetInterface
-     * @throws \RuntimeException
+     * @param array  $params - statement params
+     *
+     * @return null|\Zend\Db\ResultSet\ResultSetInterface
      */
-    public function selectWith(Select $select)
+    public function selectWith(Select $select, array $params = [])
```

Comment:
Cannot make changes to functions without making corresponding changes to TableGatewayInterface signatures, where this is defined, and all other classes implementing it, if any.